machFramework/contribute.rst

```diff
@@ -19,37 +21,54 @@ For an easier workflow, we recommend integrating these tools with your code edit
 Python
 ^^^^^^
 
-We use `black <https://github.com/psf/black>`_ for formatting Python codes.
-The version we use can be installed with:
+We use `ruff <https://github.com/astral-sh/ruff>`_ and `pre-commit <https://github.com/pre-commit/pre-commit/>`_ for linting and formatting our Python codes.
+Only pre-commit is required to run these checks locally:
 
 .. prompt:: bash
 
-    pip install black==23.1.0
+    pip install pre-commit
 
-``black`` can then be run at the project root with:
+If you want to run ruff as a standalone tool, you can install it with:
 
 .. prompt:: bash
 
-    black . -l 120
+    pip install ruff==0.12.10
+
+To match the checks we run in our CI pipelines, you will need to download our configuration files.
+Download our pre-commit configuration file from `here <https://raw.githubusercontent.com/mdolab/.github/main/.pre-commit-config.yaml>`_ and place it in the root of the respective repository, or place it somewhere central (e.g ``$HOME/.config/pre-commit``) and soft-link to it in the repository:
+
+.. prompt:: bash
+
+    mkdir -p $HOME/.config/pre-commit
+    cd $HOME/.config/pre-commit
+    wget https://raw.githubusercontent.com/mdolab/.github/main/.pre-commit-config.yaml
+    cd <path to the root of the repo>
+    ln -s $HOME/.config/pre-commit/.pre-commit-config.yaml .pre-commit-config.yaml
+
+Follow a similar process for the ruff configuration file, which can be found `here <https://raw.githubusercontent.com/mdolab/.github/main/ruff.toml>`_.
+Save the file to ``$HOME/.config/ruff/ruff.toml``.
```

Comment:
This process will not work for everyone, if they already have a `ruff` config file that they're using for repos outside of mdolab. Is there a local solution that does not involve touching user-wide global configs?